src/evaluator/evaluator/evaluator.py

```diff
@@ -0,0 +1,223 @@
+        sum_error : float = 0
+        count : int = 0
+
+        if (len(perception_output) == 0):
+            return float('inf')
+        
+        if (len(perception_ground_truth) == 0):
+            raise ValueError("No ground truth cones provided for computing average difference.")
+
+        for perception_cone in perception_output:
+            min_distance : float = np.linalg.norm(perception_cone - perception_ground_truth[0])
+
+            for ground_truth_cone in perception_ground_truth:
+                distance : float = np.linalg.norm(perception_cone - ground_truth_cone)
+                if distance < min_distance:
+                    min_distance = distance
+
+            sum_error += min_distance
+            count += 1
+        
+        average = sum_error / count
+        return average
+    
+    @staticmethod
+    def get_mean_squared_error(perception_output: list, perception_ground_truth: list):
+        """!
+        Computes the mean squared error between perception output and ground truth cones.
+
+        Args:
+            perception_output (list): List of perceived cones.
+            perception_ground_truth (list): List of ground truth cones.
+
+        Returns:
+            float: Mean squared error between perception output and ground truth cones.
+        """
+        if not perception_output:
+            raise ValueError("No perception output provided.")
+        if not perception_ground_truth:
+            raise ValueError("No ground truth cones provided.")
+
+        mse_sum = 0
+        for perception_cone in perception_output:
+            min_distance_sq = np.linalg.norm(perception_cone - perception_ground_truth[0])**2
+
+            for ground_truth_cone in perception_ground_truth:
+                distance_sq = np.linalg.norm(perception_cone - ground_truth_cone)**2
+                if distance_sq < min_distance_sq:
+                    min_distance_sq = distance_sq
+
+            mse_sum += min_distance_sq
+
+        mse = mse_sum / len(perception_output)
+        return mse
+    
+
+    @staticmethod
+    def compute_distance(cone1, cone2):
+        """!
+        Compute Euclidean distance between two cones.
+        """
+        return np.linalg.norm(cone1 - cone2)
+    
+    @staticmethod
+    def build_adjacency_matrix(cones):
+        """!
+        Build adjacency matrix based on distances between cones.
+        """
+        size = len(cones)
+        adjacency_matrix = np.zeros((size, size))
+        for i in range(size):
+            for j in range(i + 1, size):
+                distance_ij = Evaluator.compute_distance(cones[i], cones[j])
+                adjacency_matrix[i][j] = distance_ij
+                adjacency_matrix[j][i] = distance_ij
+        return adjacency_matrix
+    
+    @staticmethod
+    def get_inter_cones_distance(perception_output : list):
+        """!
+        Computes the average distance between pairs of perceived cones using Minimum Spanning Tree Prim's algorithm.
+
```

Comment:
I think scipy has a built-in implementation of Prim's algorithm, so no need for all the work 😂 but now there's no need to change that